sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/models/LinkedEntity.java

```diff
@@ -0,0 +1,177 @@
+     */
+    @JsonProperty(value = "language", required = true)
+    private String language;
+
+    /*
+     * Unique identifier of the recognized entity from the data source.
+     */
+    @JsonProperty(value = "id")
+    private String id;
+
+    /*
+     * URL for the entity's page from the data source.
+     */
+    @JsonProperty(value = "url", required = true)
+    private String url;
+
+    /*
+     * Data source used to extract entity linking, such as Wiki/Bing etc.
+     */
+    @JsonProperty(value = "dataSource", required = true)
+    private String dataSource;
+
+    /**
+     * Get the name property: Entity Linking formal name.
+     *
+     * @return the name value.
+     */
+    public String getName() {
+        return this.name;
+    }
+
+    /**
+     * Set the name property: Entity Linking formal name.
+     *
+     * @param name the name value to set.
+     * @return the LinkedEntity object itself.
+     */
+    public LinkedEntity setName(String name) {
+        this.name = name;
+        return this;
+    }
+
+    /**
+     * Get the linkedEntityMatches property: List of instances this entity appears in the
+     * text.
+     *
+     * @return the linkedEntityMatches value.
+     */
+    public List<LinkedEntityMatch> getLinkedEntityMatches() {
+        return this.linkedEntityMatches;
+    }
+
+    /**
+     * Set the linkedEntityMatches property: List of instances this entity appears in the
+     * text.
+     *
+     * @param linkedEntityMatches the linkedEntityMatches value to set.
+     * @return the LinkedEntity object itself.
+     */
+    public LinkedEntity setLinkedEntityMatches(List<LinkedEntityMatch> linkedEntityMatches) {
+        this.linkedEntityMatches = linkedEntityMatches;
+        return this;
+    }
+
+    /**
+     * Get the language property: Language used in the data source.
+     *
+     * @return the language value.
+     */
+    public String getLanguage() {
+        return this.language;
+    }
+
+    /**
+     * Set the language property: Language used in the data source.
+     *
+     * @param language the language value to set.
+     * @return the LinkedEntity object itself.
+     */
+    public LinkedEntity setLanguage(String language) {
+        this.language = language;
+        return this;
+    }
+
+    /**
+     * Get the id property: Unique identifier of the recognized entity from the
+     * data source.
+     *
+     * @return the id value.
+     */
+    public String getId() {
+        return this.id;
+    }
+
+    /**
+     * Set the id property: Unique identifier of the recognized entity from the
+     * data source.
+     *
+     * @param id the id value to set.
+     * @return the LinkedEntity object itself.
+     */
+    public LinkedEntity setId(String id) {
+        this.id = id;
+        return this;
+    }
+
+    /**
+     * Get the url property: URL for the entity's page from the data source.
+     *
+     * @return the url value.
+     */
+    public String getUri() {
```

Comment:
This should be `getUrl()`

Reply:
I think the service team requested for this to be Uri. I know we recently updated all Uri to Url, is it mentioned in our guidelines to direct to the service team?

Reply:
If we want to name this `getUri()` then setter should be named `setUri()` too and the javadocs should be updated. While URI is a superset of URL, being more specific is preferred and hence we use URL everywhere else. I am not sure if there's anything in the Java guidelines.
cc @JonathanGiles

Reply:
The setter has been removed.

Reply:
It should be url. Uri is a super set of url, where url (locator) provides a way to locate that resource, by using https:// file:// etc.

Reply:
https://azure.github.io/azure-sdk/java_design.html#java-api-url

Reply:
No, the name of the getter should be getUrl(), the return type doesn't have to change.

Reply:
updated